tee-worker/litentry/core/vc-issuance/lc-vc-task-receiver/src/lib.rs

```diff
@@ -41,20 +41,57 @@ use itp_stf_executor::traits::StfEnclaveSigning;
 use itp_stf_state_handler::handle_state::HandleState;
 use itp_top_pool_author::traits::AuthorApi;
 use itp_types::parentchain::ParentchainId;
+use lazy_static::lazy_static;
 use lc_stf_task_receiver::StfTaskContext;
 use lc_stf_task_sender::AssertionBuildRequest;
-use lc_vc_task_sender::init_vc_task_sender_storage;
+use lc_vc_task_sender::{init_vc_task_sender_storage, TrustedVCRequestSigned};
 use litentry_primitives::{
 	aes_decrypt, AesOutput, IdentityNetworkTuple, RequestAesKey, ShardIdentifier,
 };
 use std::{
+	collections::HashMap,
 	format,
 	string::{String, ToString},
 	sync::Arc,
+	time::{Duration, Instant},
 	vec::Vec,
 };
 use threadpool::ThreadPool;
 
+#[cfg(feature = "std")]
+use std::sync::RwLock;
+
+#[cfg(feature = "sgx")]
+use std::sync::SgxRwLock as RwLock;
+
+struct RateLimiter {
+	requests: RwLock<HashMap<String, Instant>>,
+}
+
+impl RateLimiter {
+	fn should_allow(&self, request_key: String) -> Result<(), String> {
+		let requests = self.requests.read().unwrap();
+		log::error!("Request key: {:?}", request_key);
+		if let Some(&last_instant) = requests.get(&request_key) {
+			if last_instant.elapsed() < Duration::from_secs(10) {
+				return Err("Request limit reached".to_string()) // Reject if within 5 minutes
+			}
+		}
+		drop(requests); // Drop read lock
+
+		log::error!("Writing to the hashmap");
+		let mut requests = self.requests.write().unwrap();
+		requests.insert(request_key, Instant::now()); // Update with current Instant
+		Ok(())
+	}
+}
+
+// Global instance using lazy_static
+lazy_static! {
+	static ref GLOBAL_RATE_LIMITER: RateLimiter =
```

Comment:
Is it possible to avoid using globals and pass `RateLimiter` as an argument to functions/object's constructors ?

Reply:
I find it easier to undestand the code and function/object's dependencies when all they come in with api arguments rather being bound to globals. 

It's also easier to test the code because you can pass whatever instance you want to object being tested. 

If you find it easier to share object through globals then it's ok, but I would like to promote using it only when really needed.

Reply:
I concur, updated to not use globals

Reply:
Yes we should avoid using globals when possible (and use [dependency injection](https://en.wikipedia.org/wiki/Dependency_injection) instead)